banzai/image/image.py

```diff
@@ -18,60 +18,42 @@
 
 class Image(object):
 
-    def __init__(self, pipeline_context, filename=None, data=None, header=None,
-                 extension_headers=None, bpm=None):
-        if header is None:
-            header = {}
-
+    def __init__(self, pipeline_context, filename: str = None, data=None, header: Header = None,
+                 extension_headers: list = None, bpm=None):
         if extension_headers is None:
             extension_headers = []
 
         if filename is not None:
             data, header, bpm, extension_headers = fits_utils.open_image(filename)
-            if '.fz' == filename[-3:]:
-                filename = filename[:-3]
+            filename = os.path.splitext(filename)[0]
+
             self.filename = os.path.basename(filename)
 
         self.data = data
-        self.header = header
+        self.header = Header(header)
         self.bpm = bpm
-
         self.extension_headers = extension_headers
-
-        self.request_number = header.get('REQNUM')
+        self.gain = self._set_gain(extension_headers, header)
         self.telescope, self.site, self.instrument = self._init_telescope_info(pipeline_context)
+        self.catalog = None
 
-        self.epoch = str(header.get('DAY-OBS'))
-        self.nx = header.get('NAXIS1')
-        self.ny = header.get('NAXIS2')
-        self.block_id = header.get('BLKUID')
-        self.molecule_id = header.get('MOLUID')
-
-        if len(self.extension_headers) > 0 and 'GAIN' in self.extension_headers[0]:
-                self.gain = [h['GAIN'] for h in extension_headers]
+    def _set_gain(self, extension_headers, header):
+        if len(extension_headers) > 0 and 'GAIN' in extension_headers[0]:
+            gain = [h['GAIN'] for h in extension_headers]
         else:
-            self.gain = eval(str(header.get('GAIN')))
+            gain = header.gain
 
-        self.ccdsum = header.get('CCDSUM')
-        self.filter = header.get('FILTER')
-
-        self.obstype = header.get('OBSTYPE')
-        self.exptime = float(header.get('EXPTIME', 0.0))
-        self.dateobs = date_utils.parse_date_obs(header.get('DATE-OBS', '1900-01-01T00:00:00.00000'))
-        self.readnoise = float(header.get('RDNOISE', 0.0))
-        self.ra, self.dec = fits_utils.parse_ra_dec(header)
-        self.pixel_scale = float(header.get('PIXSCALE', 0.0))
-        self.catalog = None
+        return gain
 
-    def _init_telescope_info(self, pipeline_context):
-        if len(self.header) > 0:
```

Comment:
When would a header ever have a zero length?

Reply:
This would be if it defaulted to an empty dict. We could do `if not self.header:` if that is better.